### ISSUE TITLE
Component with onpush requires detect changes

### ISSUE THREAD

**Issue by asafper** [AlexKhymenko/ngx-permissions#54]
Hi,

When using the ngxPermission directives inside compoent that uses OnPush strategy 
the elements are not rendered correctly, if i invoke the changeDetectorRef.detectchanges manully
than everything is fine.

The strange issue is that if i have nested levels of permission directives inside the same component
i need to invoke the changeDetectorRef.detectchanges for each nesting level.

Can you please advise how to solve this ?

**Comment by AlexKhymenko** [comment]
Can You please provide plunker of some github project example. And i will fix it :-)

**Comment by AlexKhymenko** [comment]
@asafper Cause i can't reproduce

**Comment by AlexKhymenko** [comment]
If anyone has the same problem, please create reproduction.

**Comment by jfcere** [comment]
I have the same problem in my application but I cannot reproduce it in Stackblitz... I am so confused 😕!

https://stackblitz.com/edit/ngx-permissions-onpush

### Problem with directive
When I am using `*ngxPermissionsOnlyDirective` it won't reflect immediatly in the DOM until I make an action that force change detection ...

```html
<!-- not working -->
<a *ngxPermissionsOnly="'Permission'">Link Text</a>
```

### Works with service

If I use `NgxPermissionsService.hasPermission()` and put the returned value into a property to be validated in the DOM with `*ngIf` It works like a charm ...

```html
<a *ngIf="hasPermission">Link Text</a>
```

```typescript
constructor(
  private ngxPermissionsService: NgxPermissionsService,
) {
  this.ngxPermissionsService.hasPermission('AddTenant').then(hasPermission => {
    this.hasAddTenantPermission = hasPermission;
  });
}
```

Any insight is soooo welcome ...

**Comment by jfcere** [comment]
I created my own structural directive to validate the permission and by adding `ChangeDetectionRef.markForCheck()` to trigger the change detection it fixed my  problem.

@AlexKhymenko would you find it acceptable to add `ChangeDetectionRef.markForCheck()` once the template has been instantiated/destroyed in the view container?

```html
<a *permissionsOnly="'Permission'">Link Text</a>
```

```typescript
import { ChangeDetectorRef, Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
import { NgxPermissionsService } from 'ngx-permissions';

@Directive({
  selector: '[permissionOnly]',
})
export class PermissionOnlyDirective {
  @Input()
  set permissionOnly(value: string) {
    this.ngxPermissionsService.hasPermission(value).then(hasPermission => {
      if (hasPermission && !this.hasView) {
        this.viewContainer.createEmbeddedView(this.templateRef);
        this.hasView = true;
      } else if (!hasPermission && this.hasView) {
        this.viewContainer.clear();
        this.hasView = false;
      }
      this.changeDetectorRef.markForCheck();
    });
  }

  private hasView = false;

  constructor(
    private changeDetectorRef: ChangeDetectorRef,
    private ngxPermissionsService: NgxPermissionsService,
    private templateRef: TemplateRef<any>,
    private viewContainer: ViewContainerRef,
  ) { }
}
```

**Comment by AlexKhymenko** [comment]
@jfcere Actually i was thinking about it. Just if i had repruduction it would be better. But i guess i will just add it but i can do it only on Monday

**Comment by AlexKhymenko** [comment]
@jfcere I have updated code but i didn't publish it yet. Is it possible for You to download code and verify that it fixed the problem?

**Comment by AlexKhymenko** [comment]
@jfcere publish in version 6

**Comment by AlexKhymenko** [comment]
@jfcere @asafper Fixed in version 6.0.1 If it still persist please open new issue
